keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,795 @@
+bogged down.
+-->
+Users should be able to transfer VolumeSnapshots that reference the same VolumeSnapshotContent
+across namespaces. This allows users to quickly share snapshots, utilizing data for testing
+or backups in separate namespaces.
+#### Story 1
+
+- The `prod` namespace contains a PVC that holds user information. VolumeSnapshots are taken of
+this PVC on a regular basis.
+
+- An existing VolumeSnapshot can be transferred to a test namespace. The user initiates the request
+by creating a `VolumeSnapshotTransferRequest` in the source namespace.
+
+- Once created, a `VolumeSnapshotTransferApproval` resource in the target namespace will be created.
+If matching resources are found, the VolumeSnapshot controller can init the transfer. 
+
+- When the transfer is successful, the original VolumeSnapshot should be deleted to assure there is one-to-one mapping between VolumeSnapshot and VolumeSnapshotContent.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+This KEP proposes the minimal to transfer VolumeSnapshot between namespaces, while avoiding breaking the current API.
+Many caveats have been discussed, including but not limited to :- 
+
+- [StorageClass Secrets](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html).
+- VolumeSnapshotContent.Spec.VolumeSnapshotRef is `immutable`. 
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+A `VolumeSnapshotTransferRequest` named `r` and `VolumeSnapshotTransferApproval` `a` are matched if all of the following conditions are met:
+
+- `r.metadata.name == a.spec.requestName`
+- `r.metadata.namespace == a.spec.targetNamespace`
+- `r.spec.source.namespace == a.metadata.namespace`
+- `r.spec.source.name == a.spec.source.name`
+- `r.spec.source.kind == a.spec.source.kind`
+- `r.spec.approvalName == a.metadata.name`
+
+Once matching is found, the transfer process is initiated by the controller, and it performs the following:
+
+- Creates a VolumeSnapshot in the target namespace.
+- Binds the VolumeSnapshotContent to the newly created VolumeSnapshot.
+- Deletes the source VolumeSnapshot.
```

Comment:
Please add descriptions on how to delete source `VolumeSnapshot` object without deleting `VolumeSnapshotContent` that was previously bounded to the `VolumeSnapshot`.

There will be two implementation ideas:

1. Set [`DeletionPolicy`](https://kubernetes.io/docs/concepts/storage/volume-snapshots/#delete) to `VolumeSnapshotContent` and restore it after the source `VolumeSnapshot` is deleted
   In this case, we also need to consider where to store the original `DeletionPolicy` to restore. As for [kubevirt implementation](https://docs.google.com/document/d/17H1k4lqdtJwZSjNRaQue-FhMhyk14JA_MoURpoxha5Q/edit?pli=1#bookmark=id.wjnw6tdw74ie), it is stored in `OjbectTransfer` CRD.

2. Unbind from the source `VolumeSnapshot` before deleting it, as it is done for PVC in my PoC, [here](https://github.com/mkimuram/kubernetes/commit/11b8ed04f04c7b94dd80c86b32c6cecbdd9254b4#diff-9a553e2f939f7ad1470cfe1cf8a48b56b39c6b34fb7ad820c79f2f43602499bdR1996).

Reply:
This won't be needed if we choose an approach to create another `VolumeSnapshotContent` as @xing-yang explained. For the approach, we still need to be careful that two `VolumeSnapshotContent`s are referencing a single actual backend volume before deleting the old one.  Also, we will need to be careful not to delete the actual volume when deleting either of the `VolumeSnapshotContent`s.

Reply:
I have updated it above, and using new VolumeSnapshotContent